File: factorial.py
def factorial(number):
    if number < 1:
        return 'bad input'
    if number == 1:
        return 1
    return number * factorial(number - 1)

print(factorial(4))
print(factorial(7))
print(factorial(8))
print(factorial(1))
print(factorial(0))

File: first_third/factorial.js
function factorial(num) {
    if (num < 1) {
        return 'bad input'
    }
    if (num === 1) {
        return 1
    }
    return num * factorial(num-1)
}

console.log(factorial(4))
console.log(factorial(7))
console.log(factorial(8))
console.log(factorial(1))
console.log(factorial(0))